libp2p/protocols/secure/secure.nim

```diff
@@ -120,10 +120,23 @@ method readOnce*(s: SecureConn,
   doAssert(nbytes > 0, "nbytes must be positive integer")
 
   if s.buf.data().len() == 0:
-    let buf = await s.readMessage()
+    let (buf, err) = try:
+        (await s.readMessage(), nil)
+      except CatchableError as exc:
+        (@[], exc)
+
+    if isNil(err):
```

Comment:
```suggestion
    if not isNil(err):
```

Reply:
ah! indeed, see this construct is always messing with my brain waves... I definitely don't like it to be honest, also in nim requires more instructions likely due to the `not` and likely C compiler being fooled.